### PR TITLE
implement data task

### PR DESCRIPTION
**Pull request by gvaradarajan**

### CODE REVIEW

**Comment by gvaradarajan** [comment]
NOTE: works for native, stack overflow on ESP32

**Comment by gvaradarajan** [comment]
now works on esp32, just needed slightly more stack for the main task. Waiting for approval on the general approach

File: micro-rdk/src/common/data_manager.rs [hunk 1]
```diff
@@ -165,23 +198,30 @@ where
...
+                        ..Default::default()
+                    }),
+                    sensor_contents: vec![],
                 };
+                let data: Result<Vec<SensorData>, DecodeError> = readings_to_upload
+                    .into_iter()
+                    .map(SensorData::decode)
+                    .collect();
+                upload_request.sensor_contents = data?;
+                app_client.upload_data(upload_request).await?;
```
Comment (acmorrow) [inline micro-rdk/src/common/data_manager.rs:223]: Is `readings_to_upload` held across the await? If so, could that be avoided?

**Review by acmorrow — COMMENTED** [review]
I know this is going to close, but here are my thoughts.

**Comment by gvaradarajan** [comment]
closing, this approach is now outdated

**Pull request closed without merging**